Remove __cdecl__ calling convention for [-Werror,-Wignored-attributes]

leemgs opened the pull request:
The cross build for Linux/ARM is broken while doing the compilation
by the `[-Werror,-Wignored-attributes]` policy of the cross compiler.
Let's remove `__cdecl__` calling convention from ./test/src/Interop folder.

Signed-off-by: Geunsik Lim geunsik.lim@samsung.com
Signed-off-by: Prajwal A N an.prajwal@samsung.com
Signed-off-by: MyungJoo Ham myungjoo.ham@samsung.com

jashook commented:
LGTM

benpye commented:
The pal_sxs change is good as discussed in #3533 however I am unsure if the other changes are correct, I think it would possibly break the tests on x86 as it likely relies on the cdecl convention on the managed side (assuming the test does what I think at quick glance).

jkotas commented:
Right, the change in interop tests is not good because of it would break x86.

These interop tests include https://github.com/dotnet/coreclr/blob/master/tests/src/Interop/common/xplatform.h that has some code to deal with cdecl. I think the fix should be in this header file.

jkotas commented:
cc @tijoytom

leemgs commented:
>  I think it would possibly break the tests on x86

@benpye, Thanks. I have updated the existing PR. 

> These interop tests include 
> https://github.com/dotnet/coreclr/blob/master/tests/src/Interop/common/xplatform.h that has some 
> code to deal with cdecl. 

@jkotas, I agree. I have updated the patches by using `WINAPI `of `./tests/src/Interop/common/xplatform.h` instead of the existing workaround. 

BTW, I think that ` #define WINAPI   __cdecl` more reasonable than `#define WINAPI   _cdecl` statment. Why do we have to use `_cdecl` without  `__cdecl`. If I have missed something, please correct me.

jkotas commented:
`WINAPI` is stdcall on Windows - defines by the Windows system headers. It cannot be changed to mean cdecl.

The right fix should have no deltas in LPSTRTestNative.cpp and LPTSTRTestNative.cpp. Instead, the delta will be only in xplatform.h.

jkotas commented:
My comment about just removing `__cdecl` without ifdef was meant for exceptionsxs.cpp only.

The rest of the files still need work as described above.

leemgs commented:
> It cannot be changed to mean cdecl.

I am sorry too late.  Do you mean this?  or Do you mean that we have to add another macro (`e.g., #if defined(__arm__)  with another macro `) instead of `WINAPI` macro in `tests/src/Interop/common/xplatform.h` header ?

jkotas commented:
I did not meant to introduce new macros. I meant to `#define __cdecl` in `xplatform.h` as nothing when you are not on x86. Something like:

```
#if __i386__
#define __stdcall __attribute__((stdcall))
#define _cdecl __attribute__((cdecl))
#define __cdecl __attribute__((cdecl)) <- new line
#else
#define __stdcall
#define _cdecl
#define __cdecl <- new line
#endif
```

Could you please give this a try?

leemgs commented:
@jkotas , Thanks, I have done "squash" command  to make a single commit. I have executed the below evaluation. 
```
a. For Linux/ARM: time ROOTFS_DIR=/work/dotnet/rootfs-coreclr/arm ./build.sh arm debug clean cross
b. For Linux/X64: time ./build.sh arm debug clean
```
